src/conditional.ts

```diff
@@ -1,11 +1,13 @@
 /* eslint-disable jsdoc/check-param-names, jsdoc/require-param -- we don't document the case params, it'd be redundant */
 
 import { purryOn } from "./internal/purryOn";
+import { type GuardType } from "./internal/types";
 
-type Case<In, Out, Thru extends In = In> = readonly [
-  when: ((data: In) => boolean) | ((data: In) => data is Thru),
-  then: (data: Thru) => Out,
-];
+type Case<
+  In,
+  Out,
+  CaseFn extends (x: In) => boolean = (x: In) => boolean,
```

Comment:
**thought:** i'm not sure how i feel about the name `CaseFn`

maybe `When` is better? :shrug:

Reply:
@cjquines Done